rpcs3/Emu/Cell/Modules/cellSysutil.cpp

```diff
@@ -283,18 +283,23 @@ s32 cellSysutilUnregisterCallback(u32 slot)
 	return CELL_OK;
 }
 
-s32 cellSysCacheClear(void)
+s32 cellSysCacheClear(vm::ptr<CellSysCacheParam> param)
 {
-	cellSysutil.todo("cellSysCacheClear()");
+	cellSysutil.warning("cellSysCacheClear(param=*0x%x)", param);
 
 	if (!fxm::check<CellSysCacheParam>())
 	{
 		return CELL_SYSCACHE_ERROR_NOTMOUNTED;
 	}
 
-	const std::string& local_path = vfs::get("/dev_hdd1/cache/");
+	const std::string& cache_id = param->cacheId;
+	const std::string& cache_path = "/dev_hdd1/cache/" + cache_id + '/';
+
+	if (!fs::exists(vfs::get(cache_path)) || !fs::is_dir(vfs::get(cache_path))) {
```

Comment:
more of a nit, but you're calling `vfs::get` on the same path thrice, could make that a separate local.

Reply:
I'll do this also along with the changes Megamouse proposed.